### PR TITLE
add multinomial_with_seed for sampler and test_sampler.py

### PR DESCRIPTION
**Pull request by aolemila**
## Motivation
From https://github.com/sgl-project/sglang-jax/pull/202. 
Here supports the seed feature, which can be specified by every request. But this feature has to be used with `--enable-deterministic-sampling`.

**note**
- `multinomial_with_seed` is from [SGLang codes](https://github.com/sgl-project/sglang/blob/e2ac7888b8cb1fd6c33a7ec58d27a5f5b5b24e0c/python/sglang/srt/layers/sampler.py#L268.).
- There are four large prime numbers, and based on [issue](https://github.com/sgl-project/sglang/issues/10938), we can change the numbers freely.

## Command
```bash
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache \
python3 -u -m sgl_jax.launch_server \
--model-path Qwen/Qwen3-8B \
--trust-remote-code  \
--tp-size=4 \
--device=tpu \
--mem-fraction-static=0.8 \
--chunked-prefill-size=2048 \
--download-dir=/tmp \
--dtype=bfloat16 \
--max-running-requests 256 \
--skip-server-warmup \
--page-size=128  \
--disable-radix-cache --enable-deterministic-sampling 
# Note: The `--enable-deterministic-sampling` can be enabled if you want to sample deterministically.

evalscope eval  --model Qwen-8B --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 64

python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 48 --random-input 1024 --random-output 1024 --random-range-ratio 1 --warmup-requests 0 --max-concurrency=16
```

## Accurancy

**not deterministic**

```bash
+---------+-----------+-----------------+----------+-------+---------+---------+
| Model   | Dataset   | Metric          | Subset   |   Num |   Score | Cat.0   |
+=========+===========+=================+==========+=======+=========+=========+
| Qwen-8B | gsm8k     | AverageAccuracy | main     |  1319 |  0.9045 | default |
+---------+-----------+-----------------+----------+-------+---------+---------+
```

**deterministic**
```bash
+---------+-----------+-----------------+----------+-------+---------+---------+
| Model   | Dataset   | Metric          | Subset   |   Num |   Score | Cat.0   |
+=========+===========+=================+==========+=======+=========+=========+
| Qwen-8B | gsm8k     | AverageAccuracy | main     |  1319 |  0.9113 | default |
+---------+-----------+-----------------+----------+-------+---------+---------+
```

## Benchmark

**not deterministic**
```bash
============ Serving Benchmark Result ============
Backend:                                 sgl-jax
Traffic request rate:                    inf
Max request concurrency:                 16
Successful requests:                     48
Benchmark duration (s):                  19.03
Total input tokens:                      49152
Total generated tokens:                  49152
Total generated tokens (retokenized):    49113
Request throughput (req/s):              2.52
Input token throughput (tok/s):          2583.44
Output token throughput (tok/s):         2583.44
Total token throughput (tok/s):          5166.89
Concurrency:                             15.99
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   6337.80
Median E2E Latency (ms):                 6339.11
---------------Time to First Token----------------
Mean TTFT (ms):                          156.09
Median TTFT (ms):                        161.65
P99 TTFT (ms):                           261.80
---------------Inter-Token Latency----------------
Mean ITL (ms):                           6.04
Median ITL (ms):                         5.92
P95 ITL (ms):                            6.24
P99 ITL (ms):                            6.43
Max ITL (ms):                            227.31
==================================================
```

**deterministic**

```bash
============ Serving Benchmark Result ============
Backend:                                 sgl-jax
Traffic request rate:                    inf
Max request concurrency:                 16
Successful requests:                     48
Benchmark duration (s):                  19.31
Total input tokens:                      49152
Total generated tokens:                  49152
Total generated tokens (retokenized):    49113
Request throughput (req/s):              2.49
Input token throughput (tok/s):          2545.20
Output token throughput (tok/s):         2545.20
Total token throughput (tok/s):          5090.39
Concurrency:                             15.99
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   6433.04
Median E2E Latency (ms):                 6427.44
---------------Time to First Token----------------
Mean TTFT (ms):                          161.56
Median TTFT (ms):                        165.14
P99 TTFT (ms):                           268.46
---------------Inter-Token Latency----------------
Mean ITL (ms):                           6.13
Median ITL (ms):                         6.02
P95 ITL (ms):                            6.27
P99 ITL (ms):                            6.51
Max ITL (ms):                            230.81
==================================================
```

### CODE REVIEW

File: python/sgl_jax/srt/sampling/sampling_batch_info.py [hunk 1]
```diff
@@ -110,10 +113,16 @@ def from_model_worker_batch(
                 np.array([0.0] * pad_size, dtype=batch.sampling_info.min_ps.dtype),
             ]
         )
+        padded_sampling_seed = np.concat(
```
[the remark this answers is not in the document]
Comment (aolemila) [reply python/sgl_jax/srt/sampling/sampling_batch_info.py]: ignore, it works

File: python/sgl_jax/srt/sampling/sampling_batch_info.py [hunk 2]
```diff
@@ -84,11 +88,14 @@ def from_model_worker_batch(
...
     ) -> SamplingMetadata:
+        sharding = (
+            NamedSharding(mesh, PartitionSpec()) if jax.process_count() == 1 else None
+        )
         padded_temperatures = np.concat(
             [
                 batch.sampling_info.temperatures,
                 np.array(
-                    [1.0] * pad_size, dtype=batch.sampling_info.temperatures.dtype
+                    [0.6] * pad_size, dtype=batch.sampling_info.temperatures.dtype
```
Comment (pathfinder-pf) [inline python/sgl_jax/srt/sampling/sampling_batch_info.py:98]: Why change the number? The following is the same
